src/serverdlg.cpp

```diff
@@ -52,24 +52,36 @@ CServerDlg::CServerDlg ( CServer* pNServP, CServerSettings* pNSetP, const bool b
 
     lvwClients->setAccessibleName ( tr ( "Connected clients list view" ) );
 
-    // Make My Server Public flag
-    chbRegisterServer->setWhatsThis ( "<b>" + tr ( "Make My Server Public" ) + ":</b> " +
-                                      tr ( "If the Make My Server Public check box is checked, this server registers "
-                                           "itself at the directory server so that all users of the application "
-                                           "can see the server in the connect dialog server list and "
-                                           "connect to it. The registration of the server is renewed periodically "
-                                           "to make sure that all servers in the connect dialog server list are "
-                                           "actually available." ) );
+    // Server list selection combo box
+    QString strDirectoryTypeAN = tr ( "Directory Type combo box" );
+    lblDirectoryType->setAccessibleName ( strDirectoryTypeAN );
+    cbxDirectoryType->setAccessibleName ( strDirectoryTypeAN );
+
+    QString strDirectoryTypeWT = "<b>" + tr ( "Directory" ) + ":</b> " +
+                                 tr ( "Select '%1' to leave your server unregistered." ).arg ( DirectoryTypeToString ( AT_NONE ) ) + "<br>" +
```

Comment:
```suggestion
                                 tr ( "Select '%1' to leave your Server in private mode i.e. not showing it on a server list." ).arg ( DirectoryTypeToString ( AT_NONE ) ) + "<br>" +
```

Reply:
No!!!!!   Please...

There's no such thing as "private mode".

Reply:
I think @ann0see is right at least since the term "private server" has been used so far to mean servers not listed in a directory: https://jamulus.io/wiki/Running-a-Server#server-types So we'd have to change the docs to explain what the "unregistered" server state means, and retire the use of the term "private server" - quite a big change!

@pljones statement "Even a registered server can be for private use" is interesting though if I understand correctly, since to make the term "unregistered server" work on the grounds that you can have a server registered on a custom directory that isn't public, then I think that would make a "custom directory" a "private directory" as per https://github.com/jamulussoftware/jamulus/discussions/2100 (essentially, I think explaining servers and directories as both having the property of "public" and "private" makes it easier to understand them).

Reply:
Unregistered= technical, 100% correct description. However it’s "Jamulus dev speech" and not intuitively understandable. 

Private = not 100% correct, but at least understandable and used commonly. Not only our website uses this word.

Reply:
What about using the label next to the select box to say "Not registered" and renaming "Unregistered" as Private?

Reply:
>  make a "custom directory" a "private directory" 

No!

_Some_ custom directories are private.

_All_ private directories are custom.

Can you see the difference?

Reply:
> What about using the label next to the select box to say "Not registered" and renaming "Unregistered" as Private?

You mean have a checkbox with a different label doing nothing?  And make a totally misleading cosmetic change to the dropdown?

Reply:
Basically, a server operator **must** comprehend the concept of server registration.  Yes, it's a Jamulus-specific term.  And that's because it's a Jamulus-specific concept, unfortunately.

Maybe your countries don't maintain voter registries?  You don't have to register to vote?

I think the idea of "registration" is pretty natural for a normal adult but maybe that's just my heritage.

I can see that "Unregistered" may imply some "active" sense whereas "Not registered" is passive (and thus more accurate), however.

Reply:
That does raise an interesting future project, if someone wants to do it -- adding a client whitelist to both server and directory.

Reply:
I think there’s a misunderstanding here:
In German there is a label next to the dropdown box. This says "Registered" if one selects Any Genre 1, Any genre 2,…

 if one selects "Unregistered" this label says "Unregistered" (or in German "Nicht registriert") too. IMHO: that’s waste of space. 

My suggestion:

Let the drop down say "Private" and the label say "Unregistered".

And no: in my country you don’t need to register to vote ;-). 

> I think the idea of "registration" is pretty natural for a normal adult but maybe that's just my heritage.

Yes. However I think there’s no need a musician who is running a server (without much IT knowledge) must understand the concept of Directories.

Another compromise:
What about changing Unregistered to "Don’t publish"?

Reply:
Sorry, I disagree.  You've not said anything that changes my position.  I think "register" and "directory" are totally natural.

Reply:
> That does raise an interesting future project, if someone wants to do it -- adding a client whitelist to both server and directory. Possibly lower priority than getting IPv6 support finished.

Yep. That’s somehow related to what I attempted with my eduTools fork

Reply:
Just to be clear:
> This says "Registered" if one selects Any Genre 1, Any genre 2,… if one selects "Unregistered" this label says "Unregistered" 

No, the label isn't just saying "Registered" or "Unregistered".  It's reporting the result of the attempt to registered.  If the directory is full, you'll see an error, for example, saying why your registration failed.  "Unregistered" could simply be blank, of course, if the user chooses "Not registered".  But it is _necessary_ information to the server operator when attempting to register, so it cannot be removed.

Reply:
> Not registering a server does not make it private. You could have it posted all over major world news feeds and Facebook

Totally valid.

We had a checkbox saying "Publish my server in the server list" or similar. This was quite clear.

Now we have "Not registered" in the drop down which gives a lot less information. 
Probably I‘m unhappy since the word "publish" was removed. Hence my suggestion: "Don’t publish".
Why? Unregistered doesn’t fit well to the term "private" . Private is present everywhere around Jamulus although it’s incorrect but we don’t have a fitting, non-technical term.

 I do not think we can get rid of the term "private" and introducing a new term is confusing.

Reply:
> How is "Unregistered" harder to understand than "Not registered"?

We seem to be talking about several different things. The UI change itself is not controversial as it's needed for the functionality to work I think. But this PR is also implying a possibly large change to the terminology we have been using in the documentation, and calling into question (for the first time I think) our use of terms like "private". Or have I not understood?

Put it this way: if this PR is to go ahead with the server states of "Registered" and "Unregistered", does [anything here](https://jamulus.io/wiki/Running-a-Server#server-types) need changing? It feels like it does because the reader wouldn't be able to make a direct connection between what was in the UI and what's in the docs.

Reply:
@ann0see said: We had a checkbox saying "Publish my server in the server list" or similar. This was quite clear.

Publish may be a better word than Register. I consulted a thesaurus for "publish" and found more good alternatives. "Publish" puts power in the hands of the user, whereas "Register" vaguely gets your server on some server list. Yes! ... "Publish my server in the server list" !

Reply:
@ann0see 

> Private is present everywhere around Jamulus although it’s incorrect but we don’t have a fitting, non-technical term.

Exactly. It's not a precise meaning, but it's accurate enough for the context in which we use it. Arguing against the use of "private" is not going to get us anywhere.

Reply:
> Publish may be a better word than Register. 

Perhaps, if we're going to come up with new terms for existing states of the server, then we need to be looking a the docs as well as this PR. We seem to be implying a pretty large change.

Reply:
This is the **3.8.1** UI:

"Server Setup" tab:
![image](https://user-images.githubusercontent.com/1549463/147812664-c788b81a-f228-4e58-845b-8e8becbe6a0a.png)
```
- server mode chosen
- initialization file name: ../Jamulus-wip/ini-server
- selected port number: 56881
- default directory server set: anygenre1.jamulus.io
```
On start up, the server _immediately_ registered in the default directory.  However, once the settings were loaded, it then unregistered again due to the checkbox being off.

"Make my server public"
-- This is incorrect.  Just because the checkbox is checked, does _not_ mean a server is public.
"Register my server in the Server List"
-- This was an on/off switch for registering from back in the day when there was a single directory - you registered or not - so fine.  But life moves on.

"List [DISABLED]"
-- I have to enable registration to change where I register - meaning I have to register in the _wrong_ server list before I can choose the right one!  Bad design.

"Unregistered"
-- This is the latest change in registration state -- from registered with the default directory to unregistered.  The checkbox _causes_ this status because the server _actively_ unregisters (tells the directory to remove it from the server list) successfully.

"Options" tab:
![image](https://user-images.githubusercontent.com/1549463/147813025-c95c52af-7ed2-471a-9522-ea04c1c616a0.png)
"Custom Directory Server Address"
-- is rather long winded - but at least the input box is active.

Reply:
This is the UI with this change:

Start up log:
```
- server mode chosen
- initialization file name: ini-server
- selected port number: 56881
```

"Server Setup" tab:
![image](https://user-images.githubusercontent.com/1549463/147813217-2d6597d9-0758-4770-8d9b-1e5e063527ba.png)

"Directory [Not registered]"
-- "Directory" because that's the current term we're using in the client, so this makes it consistent.  "Not registered" because "register" is the term that has been used since Jamulus first had a server list and, with this item selected, the server _does not register_ on start up at all.

"Not registered"
-- As the "result" of registration - in this case, no registration was attempted, so it makes sense.  It's going to say that if you switch from registered to no longer registered, too.

"Options" tab:
![image](https://user-images.githubusercontent.com/1549463/147813506-bf21a6c0-0a8f-46bb-9bc9-edd36b7a5cd5.png)
"Custom Directory Server Address [ ---value--- ]"
-- I've not changed it yet.  I did change it -- then had second thoughts.  This is a "Custom Directory" "server address" - i.e. the "data type" is "server address" and the "usage" is "Custom Directory", so the full text gives the best context as to what value to enter.  The "What's this?" text on the field is wrong, I just noticed...  Any UI/UX changes here, I'd rather keep separate at this stage.

"{Server List Filename} [| --- value --- |] {<X}"
--- Finally the "new" bit - allows choosing a server list file name using Qt file chooser (click button).  The "clear" button clears the value.

Reply:
> "Publish my server in the server list" 
I don't like "publish" as it conflicts with the idea of private directories and private servers registering with them.  A _published_ private server?  It seems wrong.

Reply:
![image](https://user-images.githubusercontent.com/4561747/147885636-07689829-a50d-414d-b763-89308468cbb8.png)

1. Generally, the convention with drop-downs is they describe their own state (in this case "registered/not registered"). So would it be possible to make that message only appear if the directory is full? Looks a bit odd repeated like that.

2. Not a big deal, but the convention we and others also take is that the drop-down's label is the "thing" that's being changed by the dropdown. So to the uninitiated, this looks like it's the directory that's not registered, not the server. 

----
![image](https://user-images.githubusercontent.com/4561747/147885860-9b4ff750-6f15-47b9-bb4c-b475597bbda6.png)

3. To make it look a bit more compact, you could label the address field "Custom Directory Server" and put "Enter address" into the "hint text" of the address field perhaps?

4. What does the server list file do?

----

5. On vocabulary: With "private" and "public" have gone from the UI I think it makes things less clear. I understand that Jamulus doesn't technically make anything private, and that a "public" server might be on a private directory or whatever,  but that's not the point. The terms are **accurate enough**  (shorthand, if you like) to give the _average_ user (not custom server operators, who are just going to have to read the manual) the important idea that if they make a server "public" it can be seen by others without the server operator doing anything else. And if they make it "private" then by default nobody can see it. I don't think we should be moving away from the current vocabulary because it's pretty much at the heart of Jamulus's UX.

Reply:
>     1. Generally, the convention with drop-downs is they describe their own state (in this case "[directory name]/not registered"). So would it be possible to make that message only appear if the directory is full? Looks a bit odd repeated like that.

"Successfully unregistered" might be better than "Not registered" if the server was previously registered, else blank if it hasn't been registered, as it's the result of the action taken as a result of changing the dropdown selection.  The dropdown says what state you _want_ the status label says what happened as a result of trying to achieve that for you.

> 
>     2. Not a big deal, but the convention we and others also take is that the drop-down's label is the "thing" that's being changed by the dropdown. So to the uninitiated, this looks like it's the directory that's not registered, not the server.

Yes, that's unfortunately true -- "Server registration" maybe as the label?  I wanted it to be the same as in the client for familiarity and consistency.

>     3. To make it look a bit more compact, you could label the address field "Custom Directory Server" and put "Enter address" into the "hint text" of the address field perhaps?

You mean label "Custom Directory" (no "server" on that) and "Enter server address" into the hint text.  I guess that would work.  I'll give it a go - in a separate, later change.

>     4. What does the server list file do?

Go read the PR for that change.

>     5. On vocabulary: With "private" and "public"  gone from the UI I think it makes things less clear.

We **must not** use "private".  There's no two ways about it.  It's morally wrong, technically wrong and potentially legally wrong (IANAL).

> I understand that Jamulus doesn't technically make anything private, and that a "public" server might be on a private directory or whatever,  but that's not the point. 

Hm...

> The terms are **accurate enough**

No.  They are not "accurate enough" if they are misleading people into believing something that is simply untrue.

> (shorthand, if you like) to give the _average_ user (not custom server operators, who are just going to have to read the manual) the important idea that if they make a server "public" it can be seen by others without the server operator doing anything else. And if they make it "private" then by default nobody can see it. 

> I don't think we should be moving away from the current vocabulary because it's pretty much at the heart of Jamulus's UX.

I don't believe the "current vocabulary" as you put has been in place long enough to matter.  The concepts of a directory, multiple directories and custom directories have not even been stable over the last two and a half years, so how you can come up with some idea there's a "current vocabulary" that everyone understands is just stupidly unbelievably annoying.

Forget any ideas you've decided are right "just because" and look at reality.

Reply:
> Go read the PR for that change.

Sorry, I meant how is the user supposed to know what that does? But I am a bit unclear as to the use case myself too. I guess they'd just have to read the manual, but there might be other ways of presenting it with some help text perhaps. 

> We must not use "private". There's no two ways about it.

OK, but the use of the term (also paired with "public", which I suppose is similarly imprecise by your logic) is all over the docs and in the forums. So if we are to retire them on the grounds you state, what do we say [here](https://jamulus.io/wiki/Running-a-Server#server-types) and [here](https://jamulus.io/wiki/Running-a-Server#running-a-private-server) and [elsewhere](https://github.com/jamulussoftware/jamulus/discussions?discussions_q=private)? I really don't think anyone is being "misled" as you say.

Reply:
See here:
https://github.com/jamulussoftware/jamulus/pull/2199#issuecomment-1003943082

Reply:
> > Go read the PR for that change.
> 
> Sorry, I meant how is the user supposed to know what that does? But I am a bit unclear as to the use case myself too. I guess they'd just have to read the manual, but there might be other ways of presenting it with some help text perhaps.

Right-click, "What's this?" or read the online documentation (which needs updating, of course).

Reply:
OK I'll have a look at the docs. Meanwhile, the "server list file" is used to preserve the list of servers on a directory between reboots, is that right?

Reply:
> Meanwhile, the "server list file" is used to preserve the list of servers on a directory between reboots, is that right?

Yes, the change to add persistence for the server list added the file.  That went into 3.8.1.

The biggest problem I hit here was that most of the server settings were "set and forget, restart the server with a new command line" settings... even some of the UI ones had no effect until restart...  All that will change...

Reply:
> That went into 3.8.1.

Ah yes, the `--directoryfile` option. So if you don't specify this and restart the directory, then any servers registered on it will need to reconnect (that might be a better way of explaining it perhaps?).

Reply:
Yup.  It's great :).  OK, if I take my server down for a reboot, the hosted directories are offline for about 45 seconds.  But without it, the server lists come up empty and it takes up to 15 minutes for the servers to run their regular re-registration and the list to fill.  With it, the lists are full when the directory starts up.

Reply:
So is the reason why we don't have it by default to do with not being able to easily write to the filesystem?

Reply:
No, like all new features - the reason we don't change behaviour by default is that we preserve backward compatibility.